assets/src/components/site-scan-context-provider/index.js

```diff
@@ -331,7 +332,11 @@ export function SiteScanContextProvider( {
 			} )();
 		}
 
-		return clearTimeout;
+		return () => {
+			if ( typeof clearTimeout === 'function' ) {
```

Comment:
I don't think this will ever be false because `window.clearTimeout` is a function.

Reply:
Ah, nevermind. I suppose not since `clearTimeout` was defined above in the local scope.

Reply:
You just made me realize that I did an embarrassing mistake here 🤦‍♂️

This code won't clear any timeout since it's not how timeouts are cleared in JS. The `setTimeout` returns an _identifier_ of a timeout and then you should pass that identifier to the `window.clearTimeout` function.

I'll fix it in a followup PR.

Reply:
See: #6758.